File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/model/dto/response/error/ErrorResponseDTO.java
package org.infra.reactive.form.engine.form.engine.model.dto.response.error;

import lombok.*;
import org.infra.reactive.form.engine.form.engine.security.exception.ERPHttpStatus;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class ErrorResponseDTO extends RuntimeException {

    private ERPHttpStatus errorStatus;
    private String error;
    private String errorDescription;
    private String errorTraceId;
    private String errorOn;
    private String state;
    private LocalDateTime errorAt;
    private List<ErrorResponseFieldDTO> errorFields;
    private Map<String, Object> errorData;

    public ErrorResponseDTO(String error, ERPHttpStatus errorStatus) {
        this.error = error;
        this.errorStatus = errorStatus;
    }

    public List<ErrorResponseFieldDTO> getErrorFields() {
        if (errorFields == null) {
            errorFields = new ArrayList<>();
        }
        return errorFields;
    }

    public Map<String, Object> getErrorData() {
        if (errorData == null) {
            errorData = new HashMap<>();
        }
        return errorData;
    }
}

File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Response/Error/ErrorResponseDTO.js
import {BaseModel} from "../../../../../UIFrameWork/Shared/Common/BaseModel.js";
import ErrorResponseFieldDTO from "./ErrorResponseFieldDTO.js";

export default class ErrorResponseDTO extends BaseModel {

    constructor() {
        super();
    }

    getErrorStatus() {
        return this.errorStatus;
    }

    getError() {
        return this.error;
    }

    getErrorDescription() {
        return this.errorDescription;
    }

    getErrorTraceId() {
        return this.errorTraceId;
    }

    getErrorOn() {
        return this.errorOn;
    }

    getState() {
        return this.state;
    }

    getErrorAt() {
        return this.errorAt;
    }

    getErrorFields() {
        if (this.errorFields) {
            let errorFieldsDTO = [];
            this.errorFields.forEach((errorField) => {
                let errorFieldDTO = new ErrorResponseFieldDTO();
                errorFieldDTO.applyData(errorField);
                errorFieldsDTO.push(errorFieldDTO);
            })
            return errorFieldsDTO;
        } else {
            return null;
        }
    }

    getErrorData() {
        return this.errorData;
    }
}